### ISSUE TITLE
[BUG] no-magic-number doesn't work in array of widgets

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#515]
* **Dart code metrics version:** 4.4.0
* **Dart sdk version:** 2.12.2 (stable)

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
analyzer:
  exclude:
    - "**/*.g.dart"
    - "**/*.gr.dart"
    - "**/*.freezed.dart"
    - "**/generated/**.dart"
    - "**/*.gform.dart"
    - "ios/**"
  strong-mode:
    implicit-casts: false
    implicit-dynamic: false
  errors:
    missing_required_param: warning
    missing_return: error
  plugins:
    - dart_code_metrics

dart_code_metrics:
  anti-patterns:
    - long-method
    - long-parameter-list
  metrics:
    cyclomatic-complexity: 27
    source-lines-of-code: 50
    number-of-parameters: 4
    maximum-nesting-level: 5
  metrics-exclude:
    - test/**
    - testing_api_charge/**
    - e2e/**
    - ios/**
  rules:
    - always-remove-listener
    - avoid-non-null-assertion
    - avoid-returning-widgets
    - avoid-unnecessary-setstate
    - avoid-unused-parameters
    - avoid-wrapping-in-padding
    - double-literal-format
    - newline-before-return
    - no-magic-number:
        allowed: [ 0, 0.5, 1, 1.5, 2, 10, 100 ]
    - no-boolean-literal-compare
    - prefer-match-file-name:
        exclude:
          - test/**
    - prefer-single-widget-per-file
    - prefer-trailing-comma
    - no-equal-then-else
    - binary-expression-operand-order

```

</details>

<details>
<summary>Min reproducible example</summary>

```dart
import 'package:flutter/material.dart';

class TestWidget extends StatefulWidget {
  const TestWidget({Key? key}) : super(key: key);

  @override
  State<TestWidget> createState() => _TestWidgetState();
}

class _TestWidgetState extends State<TestWidget> {
  @override
  Widget build(BuildContext context) {
    return Container(
      height: 83, // rule works
      color: Colors.red.withOpacity(0.83), // rule works
      child: SizedBox(
        height: 83, // rule works
        child: Column(
          children: [
            Container(
              height: 83,  // rule doesn't work
              color: Colors.red.withOpacity(0.83),  // rule doesn't work
              child: const SizedBox(
                height: 83,  // rule doesn't work
              ),
            ),
          ],
        ),
      ),
    );
  }
}

```
</details>

**What did you do? Please include the source code example causing the issue.**

![image](https://user-images.githubusercontent.com/33546696/138591895-27f0556f-c977-4bba-93e9-02f6876fb17a.png)
**What did you expect to happen?** all `magic numbers` are highlighted

**What actually happened?** only the first one is highlighted

**Comment by HFabi** [comment]
I tried to fix this bug https://github.com/dart-code-checker/dart-code-metrics/pull/551

**Comment by incendial** [comment]
Available in 4.7.0 🚀 big thanks to @HFabi for the fix!
